src/state-mutation.js

```diff
@@ -164,7 +164,7 @@ export const removeResourceFromState = (state, resource) => {
   const entityRelationships = resource.relationships || {};
 
   return Object.keys(entityRelationships).reduce((newState, key) => {
-    if (resource.relationships[key].data === null) {
+    if (resource.relationships[key].data === undefined) {
```

Comment:
Instead of choosing one over the other may an idea could be

`if (!resource.relationships[key].data)`

Chances are if you remove a relationship we do a updateReverseRelationship thing where we would go and put it to null, the bang would handle this better than just saying its `undefined` both cases would be covered`